ras_party/controllers/notify_gateway.py

```diff
@@ -35,25 +35,25 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
             logger.info("Notification not sent. Notify is disabled.")
             return
 
-        try:
-            notification = {
-                "emailAddress": email,
-            }
-            if personalisation:
-                notification.update({"personalisation": personalisation})
-            if reference:
-                notification.update({"reference": reference})
-
-            url = urlparse.urljoin(self.notify_url, str(template_id))
-
-            response = Requests.post(url, json=notification)
-
+        notification = {
+            "emailAddress": email,
+        }
+        if personalisation:
+            notification.update({"personalisation": personalisation})
+        if reference:
+            notification.update({"reference": reference})
+
+        url = urlparse.urljoin(self.notify_url, str(template_id))
+        response = Requests.post(url, json=notification)
+        status_code = response.status_code
+
+        if status_code == 201:
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json()["id"])
-
-        except Exception as e:
+        else:
```

Comment:
is the deleted exception never thrown? we're only checking here for a 201. but codes 200 - 399 are valid and shouldnt raise an error. i think theres a method that might be useful response.raise_for_status() or something similar?

Reply:
I disagree with this - 201 is the only valid non-error response from this endpoint and if we're considering the APIs a contract we should be willing to code that in

Reply:
a 200/300 status code here will still have created the resource in whatever service is being called and will presumably be in a valid state. this error will not roll back any successful transactions and will be a nightmare to investigate any ops issue as a result of the error thrown. If a 201 is the only valid response then there should be no need to check it and a raise_for_status should suffice.